lib/hsa/kernels/rfi_chime_time_sum.cl

```diff
@@ -60,24 +60,10 @@ rfi_chime_timesum(
     output[2 + address] = tmp.s2;
     output[3 + address] = tmp.s3;
 
-    //const uint vector_index = element_index % 4;
-
-    //Output the variance for the selected feed
-    //if(gx == element_index / 4) {
-    //    switch(vector_index) {
-    //        case 0:
-    //            output_var[gy] = power_across_time.s0;
-    //            break;
-    //        case 1:
-    //            output_var[gy] = power_across_time.s1;
-    //            break;
-    //        case 2:
-    //            output_var[gy] = power_across_time.s2;
-    //            break;
-    //        case 3:
-    //            output_var[gy] = power_across_time.s3;
-    //            break;
-    //    }
-    //}
+    //Output the variance, normalising with the number of good inputs 
+    output_var[address + 0] = power_across_time.s0 / (num_elements - num_bad_inputs);
+    output_var[address + 1] = power_across_time.s1 / (num_elements - num_bad_inputs);
+    output_var[address + 2] = power_across_time.s2 / (num_elements - num_bad_inputs);
+    output_var[address + 3] = power_across_time.s3 / (num_elements - num_bad_inputs);
```

Comment:
This is a strange place to renormalize for bad inputs since you are only summing over a few inputs here.   I would factor this out and renormalize in the next kernel after the sum in elements.